django/db/backends/oracle/schema.py

```diff
@@ -56,22 +63,35 @@ def alter_field(self, model, old_field, new_field, strict=False):
             # SQLite-ish workaround
             if 'ORA-22858' in description or 'ORA-22859' in description:
                 self._alter_field_type_workaround(model, old_field, new_field)
+            # If we're changing identity column type to non-numeric type
```

Comment:
If an identity column is changing to a non-numeric type, the drop the identity first.